File: Juego/Invasiones/fuente/GUI/MenuDeConfirmacion.cs
using System;
using System.Collections.Generic;
using System.Text;
using Invasiones.Recursos;
using Invasiones.Dibujo;

namespace Invasiones.GUI
{
	class MenuDeConfirmacion : CajaGUI
	{
		/// <summary>
		/// Boton izquierdo
		/// </summary>
		private Boton m_botonIzq;

		/// <summary>
		/// Boton derecho
		/// </summary>
		private Boton m_botonDer;


		/// <summary>
		/// Las posibles selecciones dentro del menu.
		/// </summary>
		public enum SELECCION
		{ 
			NINGUNO = -1,
			IZQUIERDO,
			DERECHO
		}

		/// <summary>
		/// Costructor.
		/// </summary>
		/// <param name="leyenda">EL título que va a tener el menú de confirmación.</param>
		/// <param name="boton1">El título del boton izquierdo.</param>
		/// <param name="boton2">El título del boton derecho.</param>
		public MenuDeConfirmacion(int leyenda, int boton1, int boton2)
		{
			m_leyenda = leyenda;
			m_botonIzq = new Boton(boton1, null);
			m_botonIzq.SetearPosicion(0, 0, 0);
			m_botonDer = new Boton(boton2, null);
			m_botonDer.SetearPosicion(200, 200, 0);
			m_ancho = Definiciones.CONFIRMACION_ANCHO;
			m_alto = Definiciones.CONFIRMACION_ALTO;
		}


		/// <summary>
		/// Setea la posicion en donde se va a pintar el menú.
		/// </summary>
		/// <param name="x">Posición x.</param>
		/// <param name="y">Posición y.</param>
		/// <param name="ancla">El ancla de donde tomar el x, y</param>
		public override void SetearPosicion(int x, int y, int ancla)
		{
			m_x = x;
			m_y = y;
			if ((ancla & Superficie.H_CENTRO) != 0)
			{ 
				m_x += (Video.Ancho >> 1) -  (m_ancho>>1);
			}

			if ((ancla & Superficie.V_CENTRO) != 0)
			{
				m_y += (Video.Alto >> 1) - (m_alto >> 1);
			}

			m_botonIzq.SetearPosicion(m_x + Boton.OFFSET_LIMITE_PANTALLA, m_y + m_alto - m_botonIzq.Alto - Boton.OFFSET_LIMITE_PANTALLA, 0);
			m_botonDer.SetearPosicion(m_x  + m_ancho  - m_botonDer.Ancho - Boton.OFFSET_LIMITE_PANTALLA, m_y + m_alto - m_botonDer.Alto - Boton.OFFSET_LIMITE_PANTALLA, 0);

			

			//m_botonIzq.SetearPosicion(
		}


		/// <summary>
		/// Dibuja el boton.
		/// </summary>
		/// <param name="g">El Video en donde pintar.</param>
		public override void Dibujar(Video g)
		{
			g.SetearColor(Definiciones.GUI_COLOR_MENUS);
			g.LlenarRectangulo(m_x, m_y, m_ancho, m_alto, Definiciones.CONFIRMACION_ALPHA, 0);

			g.SetearFuente(AdministradorDeRecursos.Instancia.Fuentes[Definiciones.FUENTE_MENU], Definiciones.GUI_COLOR_TEXTO);

			g.Escribir(m_leyenda, m_x - (Video.Ancho >> 1) + (m_ancho >> 1), m_y + m_alto / 5, Superficie.H_CENTRO);

			m_botonIzq.Dibujar(g);
			m_botonDer.Dibujar(g);
		}

		/// <summary>
		/// Actualiza el menú.
		/// </summary>
		/// <returns>SELECCION.IZQUIERDO si se apreto el botón izquierdo, 
		/// SELECCION.DERECHO si se apreto el botón derecho, 
		/// SELECCION.NINGUNO si no se apreto ningun botón. </returns>
		public override int Actualizar()
		{
			if (m_botonIzq.Actualizar() != 0)
			{
				return (int)SELECCION.IZQUIERDO;
			}

			if (m_botonDer.Actualizar() != 0)
			{
				return (int)SELECCION.DERECHO;
			}

			return (int) SELECCION.NINGUNO;
		}
	}
}

File: macOS/Invasiones/Invasiones/GUI/MenuDeConfirmacion.swift
//
//  MenuDeConfirmacion.swift
//  Invasiones
//
//  Created by Lucia Medina Fretes on 06.04.26.
//
//  Port of MenuDeConfirmacion.cs — dialog box with two buttons.
//

import Foundation

class MenuDeConfirmacion: CajaGUI {

    // MARK: - Enum
    enum SELECCION: Int {
        case NINGUNO   = -1
        case IZQUIERDO =  0
        case DERECHO   =  1
    }

    // MARK: - Declarations
    private var m_botonIzq: Boton
    private var m_botonDer: Boton

    // MARK: - Initializer
    init(_ leyenda: Int, _ boton1: Int, _ boton2: Int) {
        m_botonIzq = Boton(leyenda: boton1, fuente: nil)
        m_botonIzq.setearPosicion(0, 0, 0)
        m_botonDer = Boton(leyenda: boton2, fuente: nil)
        m_botonDer.setearPosicion(200, 200, 0)
        super.init()
        m_leyenda = leyenda
        m_ancho   = Definiciones.CONFIRMACION_ANCHO
        m_alto    = Definiciones.CONFIRMACION_ALTO
    }

    // MARK: - CajaGUI overrides

    override func setearPosicion(_ x: Int, _ y: Int, _ ancla: Int) {
        m_x = x
        m_y = y
        if (ancla & Superficie.H_CENTRO) != 0 { m_x += (Video.Ancho >> 1) - (m_ancho >> 1) }
        if (ancla & Superficie.V_CENTRO) != 0 { m_y += (Video.Alto  >> 1) - (m_alto  >> 1) }

        m_botonIzq.setearPosicion(
            m_x + Boton.OFFSET_LIMITE_PANTALLA,
            m_y + m_alto - m_botonIzq.alto - Boton.OFFSET_LIMITE_PANTALLA,
            0)
        m_botonDer.setearPosicion(
            m_x + m_ancho - m_botonDer.ancho - Boton.OFFSET_LIMITE_PANTALLA,
            m_y + m_alto  - m_botonDer.alto  - Boton.OFFSET_LIMITE_PANTALLA,
            0)
    }

    @discardableResult
    override func actualizar() -> Int {
        if m_botonIzq.actualizar() != 0 { return SELECCION.IZQUIERDO.rawValue }
        if m_botonDer.actualizar() != 0 { return SELECCION.DERECHO.rawValue   }
        return SELECCION.NINGUNO.rawValue
    }

    override func dibujar(_ g: Video) {
        g.setearColor(Definiciones.GUI_COLOR_MENUS)
        g.llenarRectangulo(m_x, m_y, m_ancho, m_alto, Definiciones.CONFIRMACION_ALPHA)

        g.setearFuente(AdministradorDeRecursos.Instancia.fuentes[Definiciones.FUENTE_MENU],
                       Definiciones.GUI_COLOR_TEXTO)
        g.escribir(m_leyenda,
                   m_x - (Video.Ancho >> 1) + (m_ancho >> 1),
                   m_y + m_alto / 5,
                   Superficie.H_CENTRO)

        m_botonIzq.dibujar(g)
        m_botonDer.dibujar(g)
    }
}
